server/src/test/java/org/elasticsearch/cluster/routing/PrimaryAllocationIT.java

```diff
@@ -309,4 +320,68 @@ public void testForceAllocatePrimaryOnNoDecision() throws Exception {
         assertEquals(1, client().admin().cluster().prepareState().get().getState()
                             .routingTable().index(indexName).shardsWithState(ShardRoutingState.STARTED).size());
     }
+
+    /**
+     * This test asserts that replicas failed to execute resync operations will be failed but not marked as stale.
+     */
+    public void testPrimaryReplicaResyncFailed() throws Exception {
+        String master = internalCluster().startMasterOnlyNode(Settings.EMPTY);
+        final int numberOfReplicas = between(2, 3);
+        assertAcked(
+            prepareCreate("test", Settings.builder().put(indexSettings())
```

Comment:
start the data node first? There's no need to wait on yellow then (index creation waits for primaries to be allocated by default).

Reply:
what about this comment?

Reply:
@ywelsch Sorry, I did not see your second comment. This was addressed but the change (eg. removing ensureYellow) was several lines away from the comment, GH did not pick it up.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/28534/commits/0329f5b3f85b590913aa42d3431f3834a32f8774